scripts/class_balance.py

```diff
@@ -19,14 +19,14 @@ class Args(Tap):
     split_type: Literal['random', 'scaffold'] = 'scaffold'  # Split type, either "random" or "scaffold"
 
 
-def class_balance(data_path: str, split_type: str):
+def class_balance(data_path: str, split_type: str, args: Args):
```

Comment:
`args` is currently a global variable in this module. I think making it an argument is a good idea, but it would be best to remove the global variable definition below.

In the `if __name__ == '__main__'` section, You could pass `Args().parse_args()` directly to the `class_balance` function instead of storing it in `args` first.

Reply:
This is a good suggestion, thanks

Reply:
This is a good suggestion! Thanks!

Reply:
Yeah, the first hint was that the script (presumably) worked before you passed args as an argument, even though args was used in the `class_balance` function. The second is that PyCharm flags the argument name as shadowing a name from the outer scope.

Any variable defined at the module level is automatically global, and _using_ a global variable is also automatic if it does not exist within the function scope. The `global` keyword is only needed to "export" a local variable to the global scope, e.g. when _setting_ a global variable from inside a function.